core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -302,8 +336,7 @@ public void transportTerminated() {
           log.warning("transportTerminated called without previous transportShutdown");
           activeTransport = null;
         }
-        // TODO(notcarl): replace this with something more meaningful
-        transportShutdown(Status.UNKNOWN);
+        clearActiveTransport();
```

Comment:
Why no longer call `transportShutdown()`? Note that `clearActiveTransport()` is guaranteed to be a noop here, because of the condition at the beginning of the method.